jackrabbit-spi2dav/src/main/java/org/apache/jackrabbit/spi2dav/RepositoryServiceImpl.java

```diff
@@ -529,16 +557,49 @@ private static Object getClientKey(SessionInfo sessionInfo) {
     }
 
     protected HttpClient getClient(SessionInfo sessionInfo) throws RepositoryException {
+        return this.getClient(sessionInfo, this.allowInsecureHttps);
+    }
+
+    protected HttpClient getClient(SessionInfo sessionInfo, boolean allowInsecureHttps) throws RepositoryException {
         Object clientKey = getClientKey(sessionInfo);
         HttpClient client = clients.get(clientKey);
         if (client == null) {
-            client = httpClientBuilder.build();
-            if (sessionInfo != null) {
-                checkSessionInfo(sessionInfo);
-                clients.put(clientKey, client);
-                log.debug("Created Client " + client + " for SessionInfo " + sessionInfo);
+            try {
+                if (allowInsecureHttps) {
+                    // use the TrustSelfSignedStrategy to allow Self Signed Certificates
+                    log.info("allowInsecureHttps flag is true, enabling self signed strategy");
+                    SSLContext sslContext = SSLContextBuilder
+                            .create()
+                            .loadTrustMaterial(new TrustSelfSignedStrategy())
+                            .build();
+
+                    // we can optionally disable hostname verification.
+                    HostnameVerifier allowAllHosts = new NoopHostnameVerifier();
+
+                    // create an SSL Socket Factory to use the SSLContext with the trust self signed certificate strategy
+                    // and allow all hosts verifier.
```

Comment:
what is the use case for this? Can we rather restrict that to accepting only self-signed certificates?